common/pycontainers.h

```diff
@@ -257,36 +249,34 @@ template <typename T1, typename T2, typename value_conv> struct map_pair_wrapper
 
     struct pair_iterator_wrapper
     {
-        static object next(std::pair<wrapped_pair &, int> &iter)
+        static py::object next(std::pair<wrapped_pair &, int> &iter)
         {
             if (iter.second == 0) {
                 iter.second++;
-                return object(PythonConversion::string_converter<decltype(iter.first.base.first)>().to_str(
+                return py::cast(PythonConversion::string_converter<decltype(iter.first.base.first)>().to_str(
                         iter.first.ctx, iter.first.base.first));
             } else if (iter.second == 1) {
                 iter.second++;
-                return object(value_conv()(iter.first.ctx, iter.first.base.second));
+                return py::cast(value_conv()(iter.first.ctx, iter.first.base.second));
             } else {
                 PyErr_SetString(PyExc_StopIteration, "End of range reached");
-                boost::python::throw_error_already_set();
-                // Should be unreachable, but prevent control may reach end of
-                // non-void
-                throw std::runtime_error("unreachable");
+                throw py::error_already_set();
             }
         }
 
-        static void wrap(const char *python_name)
+        static void wrap(py::module &m, const char *python_name)
         {
-            class_<std::pair<wrapped_pair &, int>>(python_name, no_init).def("__next__", next);
+            //FIXME
+            //py::class_<std::pair<wrapped_pair &, int>>(m, python_name).def("__next__", next);
```

Comment:
@daveshah1 This one needs some thinking. Have to check if it can be done in some other way

Reply:
What is the problem?

Reply:
src/nextpnr/3rdparty/pybind11/include/pybind11/cast.h:1982:51: error: cannot bind non-const lvalue reference of type ‘std::pair<nextpnr_ice40::PythonConversion::ContextualWrapper<std::pair<const nextpnr_ice40::IdString, nextpnr_ice40::HierarchicalCell>&>&, int>&’ to an rvalue of type ‘pybind11::detail::tuple_caster<std::pair, nextpnr_ice40::PythonConversion::ContextualWrapper<std::pair<const nextpnr_ice40::IdString, nextpnr_ice40::HierarchicalCell>&>&, int>::cast_op_type<std::pair<nextpnr_ice40::PythonConversion::ContextualWrapper<std::pair<const nextpnr_ice40::IdString, nextpnr_ice40::HierarchicalCell>&>&, int>&>’ {aka ‘std::pair<nextpnr_ice40::PythonConversion::ContextualWrapper<std::pair<const nextpnr_ice40::IdString, nextpnr_ice40::HierarchicalCell>&>&, int>’}
 1982 |         return std::forward<Func>(f)(cast_op<Args>(std::move(std::get<Is>(argcasters)))...);

This is actual error when this is enabled.